lib/sgov/sgov-adapter.ts

```diff
@@ -0,0 +1,227 @@
+import {HttpFetch} from "../io/fetch/fetch-api";
+import {RdfSource, RdfSourceWrap} from "../core/adapter/rdf";
+import {isSgovClass, loadSgovClass} from "./entity-adapters/sgov-class-adapter";
+import {CoreResource, ReadOnlyMemoryStore} from "../core";
+import {FederatedSource} from "../io/rdf/federated/federated-rdf-source";
+import {RDFS, SKOS} from "./sgov-vocabulary";
+import {isSgovAssociation, loadSgovAssociation} from "./entity-adapters/sgov-association-adapter";
+import {isSgovAttribute, loadSgovAttribute} from "./entity-adapters/sgov-attribute-adapter";
+import {PimClass} from "../pim/model";
+
+const getSurroundings = [
+  getSurroundingsAttributes,
+  getSurroundingsInwardsRelations,
+  getSurroundingsOutwardsRelations,
+  getSurroundingsParent,
+  getSurroundingsComplexAttributes,
+];
+
+const compressQueryString = (query: string): string => {
+  return query.replace(/^\s*#.*$/mg, "").replace(/[ \t]+/g, " ");
+};
+
+const searchQuery = (searchString: string) => compressQueryString(search.replace("%QUERY%", `"${jsStringEscape(searchString)}"`));
+const getClassQuery = (cimIri: string) => compressQueryString(getClass.replace(/%NODE%/g, `<${cimIri}>`));
+const getSurroundingsQueries = (cimIri: string) => getSurroundings.map(q => compressQueryString(q.replace(/%NODE%/g, `<${cimIri}>`)));
+const getHierarchyQuery = (cimIri: string) => compressQueryString(getHierarchy.replace(/%NODE%/g, `<${cimIri}>`));
+const getGroupQuery = (cimIri: string) => compressQueryString(getGroup.replace(/%NODE%/g, `<${cimIri}>`));
+
+const IRI_REGEXP = new RegExp("^https?://");
+
+/**
+ * Adapter for SGOV (Semantic Government Vocabulary) CIM type.
+ *  - https://slovník.gov.cz/sparql
+ */
+export class SgovAdapter implements CimAdapter {
+  protected readonly endpoint: string;
+  protected readonly httpFetch: HttpFetch;
+  protected iriProvider: IriProvider;
+
+  protected resourceGroupCache = new Map<string, string[] | undefined>();
+
+  constructor(endpoint: string, httpFetch: HttpFetch) {
+    this.endpoint = endpoint;
+    this.httpFetch = httpFetch;
+  }
+
+  setIriProvider(iriProvider: IriProvider): void {
+    this.iriProvider = iriProvider;
+  }
+
+  async search(searchString: string): Promise<PimClass[]> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, searchQuery(searchString));
+    await source.query();
+
+    const results = await source.property("__search_results", "__has_search_result");
+
+    let sorted = [];
+    for (const result of results) {
+      const resultWrap = RdfSourceWrap.forIri(result.value, source);
+      await this.cacheResourceGroup(resultWrap);
+      sorted.push({
+        sort: Number((await resultWrap.property("__order"))[0].value),
+        cls: await loadSgovClass(resultWrap, this.iriProvider),
+      });
+    }
+    sorted = sorted.sort((a, b) => a.sort - b.sort).map(p => p.cls);
```

Comment:
I like to use ```left```, ```right``` in this scenarios. Also instead of ```sort``` I think the better name, based on the usage, is ```order``` or ```priority```.